src/rgw/rgw_admin.cc

```diff
@@ -1222,6 +1222,26 @@ static int init_bucket(const string& tenant_name, const string& bucket_name, con
   return 0;
 }
 
+bool find_bucket_by_id(CephContext *cct, RGWBucketAdminOpState& bucket_op, list<string>& keys, const string& bucket_id)
```

Comment:
some comments about this helper function that could make it more generally useful outside of radosgw-admin:

the name `find_bucket_by_id()` would imply that its searching all buckets, but this function is only searching the subset of `keys` passed in by the caller. so i would recommend moving the entire loop (`list_keys_init()` / `list_keys_next()` / `list_keys_complete()`) into this function

the function can take `RGWMetadataManager *mgr` as an argument, so it doesn't have to repeat the `store->ctl()->meta.mgr` bits in each of those calls

there's also the dependency on `RGWBucketAdminOpState`, which would make it hard to use this function elsewhere in radosgw. instead, it could use `rgw_bucket* bucket` as an output parameter, and the caller could do the `bucket_op.set_tenant()/set_bucket_name()` parts itself

i would also rename this function to `rgw_find_bucket_by_id()`, move its body to rgw_bucket.cc, and declare its prototype in rgw_bucket.h, ie:
```c++
bool rgw_find_bucket_by_id(CephContext *cct, RGWMetadataManager *mgr,
                           const string& bucket_id, rgw_bucket* bucket_out);
```

that would make the part in rgw_admin.cc much simpler:
```c++
  if (bucket_name.empty() && !bucket_id.empty()) {
    rgw_bucket bucket;
    if (!rgw_find_bucket_by_id(store->ctx(), store->ctl()->meta.mgr, bucket_id, &bucket)) {
      cerr << "failure: no such bucket id" << std::endl;
      return -ENOENT;
    }
    bucket_op.set_tenant(bucket.tenant);
    bucket_op.set_bucket_name(bucket.name);
  }
  ...
```

Reply:
Thank you, @cbodley for the feedback. As suggested I have made the changes.